components/ingest-service/pipeline/processor/action_project_tagger_test.go

```diff
@@ -0,0 +1,410 @@
+			action: &chef.Action{
+				RemoteHostname:   "chef_server_2",
+				OrganizationName: "org_1",
+			},
+			rules: []*iam_v2.ProjectRule{
+				{
+					Type: iam_v2.ProjectRuleTypes_EVENT,
+					Conditions: []*iam_v2.Condition{
+						{
+							Type:   iam_v2.ProjectRuleConditionTypes_CHEF_SERVERS,
+							Values: []string{"chef_server_1"},
+						},
+					},
+				},
+			},
+		},
+		{
+			description: "Chef Server: two condition values on the same field one matching",
+			matching:    true,
+			action: &chef.Action{
+				RemoteHostname:   "chef_server_1",
+				OrganizationName: "org_1",
+			},
+			rules: []*iam_v2.ProjectRule{
+				{
+					Type: iam_v2.ProjectRuleTypes_EVENT,
+					Conditions: []*iam_v2.Condition{
+						{
+							Type:   iam_v2.ProjectRuleConditionTypes_CHEF_SERVERS,
+							Values: []string{"chef_server_1", "chef_server_2"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Run all the cases!
+	for _, test := range cases {
+		t.Run(fmt.Sprintf("action match: %s", test.description),
+			func(t *testing.T) {
+				projectMatch := actionMatchesRules(test.action, test.rules)
+
+				assert.Equal(t, test.matching, projectMatch, test.description)
+			})
+	}
+}
+
+func TestActionBundlerSingleMessage(t *testing.T) {
+	inbox := make(chan message.ChefAction, 100)
+	listProjectRulesCount := 0
+	authzClient := iam_v2.NewMockProjectsClient(gomock.NewController(t))
+	authzClient.EXPECT().ListProjectRules(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx interface{}, in interface{}) (*iam_v2.ProjectCollectionRulesResp, error) {
+			listProjectRulesCount++
+			return &iam_v2.ProjectCollectionRulesResp{}, nil
+		})
+	errc := make(chan error)
+
+	inbox <- message.NewChefAction(context.Background(), &chef.Action{}, errc)
+	close(inbox)
+	out := actionBundleProjectTagger(inbox, authzClient)
+
+	<-out
+
+	assert.Equal(t, 1, listProjectRulesCount)
+}
+
+// When 5 messages are in the inbox the ListProjectRules function is only called once.
+func TestActionBundler5Messages(t *testing.T) {
+	inbox := make(chan message.ChefAction, 100)
+	authzClient := iam_v2.NewMockProjectsClient(gomock.NewController(t))
+	authzClient.EXPECT().ListProjectRules(gomock.Any(), gomock.Any()).Times(1).Return(
+		&iam_v2.ProjectCollectionRulesResp{}, nil)
+	errc := make(chan error)
+
+	inbox <- message.NewChefAction(context.Background(), &chef.Action{}, errc)
+	inbox <- message.NewChefAction(context.Background(), &chef.Action{}, errc)
+	inbox <- message.NewChefAction(context.Background(), &chef.Action{}, errc)
+	inbox <- message.NewChefAction(context.Background(), &chef.Action{}, errc)
+	inbox <- message.NewChefAction(context.Background(), &chef.Action{}, errc)
+	close(inbox)
+
+	out := actionBundleProjectTagger(inbox, authzClient)
+
+	<-out
+	<-out
+	<-out
+	<-out
+	<-out
+}
+
+// A simple run through of the bundle project tagger processor.
+// Two messages are sent through with only one matching a project.
+func TestActionBundlerMatchProjectRule(t *testing.T) {
```

Comment:
This has mostly been tested with this test here https://github.com/chef/automate/pull/214/files#diff-7e0662f4628cc26ca7731861f082208fR20, but I wanted to do a whole run through of the entire processor.